substrate/frame/aura/src/lib.rs

```diff
@@ -230,6 +230,11 @@ impl<T: Config> Pallet<T> {
 		}
 	}
 
+	/// Return current authorities length.
+	pub fn authorities_len() -> usize {
+		Authorities::<T>::decode_len().unwrap_or(0)
```

Comment:
If `decode_len` returns `None`, it's likely because there are inconsistencies in storage and the value itself won't be able to decode. My first thought was to return an error, but I see the pallet doesn't have one configured and throughout the code `panic` and `assert` are used. So we could do `expect` here or `unwrap_or(T::MaxAuthorities::get())` to be more defensive.

Reply:
If you can not decode the authorities, the length would be `0`. I don't think this is wrong.